presto-analyzer/src/main/java/com/facebook/presto/sql/analyzer/MetadataResolver.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.sql.analyzer;
+
+import com.facebook.presto.common.CatalogSchemaName;
+import com.facebook.presto.common.QualifiedObjectName;
+import com.facebook.presto.common.ViewDefinition;
+import com.facebook.presto.common.predicate.TupleDomain;
+import com.facebook.presto.spi.ColumnMetadata;
+import com.facebook.presto.spi.MaterializedViewDefinition;
+import com.facebook.presto.spi.MaterializedViewStatus;
+
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * Metadata resolver provides required metadata for analyzer functionality
+ */
+public interface MetadataResolver
+{
+    boolean catalogExists(String catalogName);
+
+    boolean schemaExists(CatalogSchemaName schema);
+
+    boolean tableExists(QualifiedObjectName tableName);
+
+    List<ColumnMetadata> getColumns(QualifiedObjectName tableName);
```

Comment:
So a table that doesn't exist returns an empty list here? same a a table with no columns that does exist?

Reply:
If the table does not exist, it would throw exception.

Reply:
That arguably should be a checked exception. Whether it is or not, it definitely  needs explicit Javadoc on this point: what exception is thrown under what circumstances